lib/util/cache/package/sqlite.ts

```diff
@@ -0,0 +1,137 @@
+import { promisify } from 'node:util';
+import zlib, { constants } from 'node:zlib';
+import Sqlite from 'better-sqlite3';
+import type { Database, Statement } from 'better-sqlite3';
+import * as upath from 'upath';
+import { logger } from '../../../logger';
+import { ensureDir } from '../../fs';
+
+const brotliCompress = promisify(zlib.brotliCompress);
+const brotliDecompress = promisify(zlib.brotliDecompress);
+
+function compress(input: unknown): Promise<Buffer> {
+  const jsonStr = JSON.stringify(input);
+  return brotliCompress(jsonStr, {
+    params: {
+      [constants.BROTLI_PARAM_MODE]: constants.BROTLI_MODE_TEXT,
+      [constants.BROTLI_PARAM_QUALITY]: 3,
+    },
+  });
+}
+
+async function decompress<T>(input: Buffer): Promise<T> {
+  const buf = await brotliDecompress(input);
+  const jsonStr = buf.toString('utf8');
+  return JSON.parse(jsonStr) as T;
+}
```

Comment:
can't we share this between redis and sqlite?

Reply:
Yes, but later